### PR TITLE
Use `cargo fmt` instead of `rustfmt`

### PR DESCRIPTION
**Pull request by mathieu-lemay**

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
 -   id: fmt
     name: fmt
-    description: Format files with rustfmt.
-    entry: rustfmt --
+    description: Format files with cargo fmt.
+    entry: cargo fmt --
```
Comment (rambo) [inline .pre-commit-hooks.yaml:4]: Is the couble-dash actually needed ? I think is will make it hard to pass args to the command should someone wisth to do it. Ref #7

Comment (mathieu-lemay) [reply .pre-commit-hooks.yaml:4]: Anything that goes after the double-dash is actually then sent to rustfmt. But you're right, adding the double-dash prevents the user to sending arguments to cargo fmt. I'll remove it.

Comment (mathieu-lemay) [reply .pre-commit-hooks.yaml:4]: Guess I should have tested it first, turns out removing the double dash actually breaks the hook. They are needed for pre-commit to be able to pass file paths.

**Comment by gilescope** [comment]
If you use `rustfmt` with `pass_filenames: true` then it's significantly faster than `cargo fmt` as it works only on the changed files. For large codebases it's a vast difference.

**Comment by rambo** [comment]
@gilescope do you have a working example configuration. I don't  actually remember anymore why I agreed on using `cargo fm` vs `rustfmt`. Probably because rustfmt maybe did not read editition from cargo.toml requiring duplicating that info in rustfmt.toml as well...

**Comment by gilescope** [comment]
Was proprietary code. I would think most code these days was targeted at
the latest edition, and configuring rustfmt.toml if not the case seems
reasonable. Much more likely that rust projects get bigger over time. This
one was >50k lines and at that point cargo fmt all the files took a few
seconds where as rustfmt a few files took no time.

**Pull request merged**